core/dbt/clients/jinja.py

```diff
@@ -285,6 +292,13 @@ def __init__(
         self.node = node
         self.stack = stack
 
+        if self.context and "adapter" in self.context:
+            if "materialization" in self.macro.unique_id:
+                self.context["special_functions"] = self.context["adapter"].submit_python_job
```

Comment:
@gshank  This way user actually still can use this submit_python_job function  via `{{special_function(model, python_code)}}` in macro, is there a way to actually have this in context but not allow jinja to access it?

Other ways I have been trying to do is save it somewhere else individually, but haven't found a good place, the `MacroGenerater` object for `materialization macro` and `statement macro` are actually two different objects, with only context being passed from one to another(node, stack are both None for `materialization macro`)

Reply:
Also tag @jtcohen6 to see if you have any thoughts